discord-bot/src/index.ts

```diff
@@ -0,0 +1,202 @@
+    })
+  }
+})
+
+client.login(token)
+console.log('Logged in')
+
+process.stdout.write('Refreshing slash commands... ')
+try {
+  await rest.put(Routes.applicationCommands(clientId), {
+    body: commands.mapValues((c) => c.data.toJSON()),
+  })
+
+  console.log(
+    `${[...commands.values()].map((x) => '/' + x.data.name).join(', ')}`
+  )
+} catch (error) {
+  console.log('ERROR')
+  console.error(error)
+}
+const handleOldReaction = async (
+  pReaction: MessageReaction | PartialMessageReaction,
+  pUser: User | PartialUser
+) => {
+  const ignore = messagesHandledViaInteraction.has(pReaction.message.id)
+  console.log(`ignoring reaction:${ignore}`)
+  if (ignore) return
+
+  console.log('checking old reaction for proper details')
+
+  const reaction = pReaction.partial ? await pReaction.fetch() : pReaction
+  console.log('got reaction emoji id', reaction.emoji.id)
+  console.log('message is partial:', reaction.message.partial)
+  const message = reaction.message.partial
+    ? await reaction.message.fetch()
+    : reaction.message
+
+  console.log('got old message id', message.id)
+  const emojiKey = getAnyHandledEmojiKey(reaction)
+  console.log('got emoji key', emojiKey)
+  if (!emojiKey) return
+
+  const user = pUser.partial ? await pUser.fetch() : pUser
+
+  const channel = await client.channels.fetch(message.channelId)
+  console.log('got channel', channel?.id)
+  if (!channel || !channel.isTextBased()) return
+  console.log('message embeds: ', message.embeds.length)
+  const fetchedMessage = await reaction.message.fetch()
+  console.log('fetched message embeds:', fetchedMessage.embeds.length)
+
+  const marketEmbed = EmbedBuilder.from(message.embeds[0])
+  console.log('got market embed json:', marketEmbed.toJSON())
+  const link = marketEmbed.toJSON().url
+  console.log('got link', link)
+  if (!link || !link.startsWith('https://manifold.markets/')) return
+
+  const slug = getSlug(link)
+  console.log('got slug', slug)
+  if (!slug) {
+    await sendThreadErrorMessage(
+      channel as TextChannel,
+      `Slug-not-found-error-${link}`,
+      'Could not find market slug in message',
+      user as User
+    )
+    return
+  }
+
+  const market = await getMarketFromSlug(slug, (error) =>
+    sendThreadErrorMessage(
+      channel as TextChannel,
+      `Error-${slug}`,
+      error,
+      user as User
+    )
+  )
+  console.log('got market url', market?.url)
+  if (!market) return
+  if (!(await userApiKey(user.id)))
+    await Promise.all(message.reactions.cache?.map((r) => r.users.fetch()))
+
+  await handleReaction(reaction, user, channel as TextChannel, market)
+}
+
+client.on(Events.MessageReactionAdd, handleOldReaction)
+
+client.on(Events.MessageCreate, async (message) => {
+  // Here you check for channel type
+  // We only need direct messages here, so skip other messages
+  if (!message.channel.isDMBased()) return
+  if (message.author.id === client.user?.id) return
+  await registerApiKey(message)
```

Comment:
It's kind of a shame that it will assume anyone talking to it is sending an API key and produce a correspondingly weird error message if someone tries to e.g. use the /market command via DM.

Reply:
We can add more functionalities for DMing the bot later. I don't want to force the user to use a slash command to register their key as they feel pretty foreign, at least to me.